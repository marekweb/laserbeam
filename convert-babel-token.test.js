const convertBabelTokens = require("./convert-babel-tokens");
const parse = require("./parse");

const source = `const a = [f("b" + 1)].x[1];`;

it("produces an array of converted tokens", () => {
  const input = parse(source);
  const convertedTokens = convertBabelTokens(input.tokens, source);
  expect(convertedTokens).toMatchSnapshot();
});
