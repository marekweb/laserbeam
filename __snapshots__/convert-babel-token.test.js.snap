// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`produces an array of converted tokens 1`] = `
Array [
  Object {
    "end": 5,
    "start": 0,
    "text": "const",
    "type": "keyword",
    "value": "const",
  },
  Object {
    "end": 6,
    "start": 5,
    "text": " ",
    "type": "whitespace",
  },
  Object {
    "end": 7,
    "start": 6,
    "text": "a",
    "type": "name",
    "value": "a",
  },
  Object {
    "end": 8,
    "start": 7,
    "text": " ",
    "type": "whitespace",
  },
  Object {
    "end": 9,
    "start": 8,
    "text": "=",
    "type": "punctuation",
    "value": "=",
  },
  Object {
    "end": 10,
    "start": 9,
    "text": " ",
    "type": "whitespace",
  },
  Object {
    "end": 11,
    "start": 10,
    "text": "[",
    "type": "punctuation",
    "value": "[",
  },
  Object {
    "end": 12,
    "start": 11,
    "text": "f",
    "type": "name",
    "value": "f",
  },
  Object {
    "end": 13,
    "start": 12,
    "text": "(",
    "type": "punctuation",
    "value": "(",
  },
  Object {
    "end": 16,
    "start": 13,
    "text": "\\"b\\"",
    "type": "string",
    "value": "b",
  },
  Object {
    "end": 17,
    "start": 16,
    "text": " ",
    "type": "whitespace",
  },
  Object {
    "end": 18,
    "start": 17,
    "text": "+",
    "type": "punctuation",
    "value": "+",
  },
  Object {
    "end": 19,
    "start": 18,
    "text": " ",
    "type": "whitespace",
  },
  Object {
    "end": 20,
    "start": 19,
    "text": "1",
    "type": "num",
    "value": "1",
  },
  Object {
    "end": 21,
    "start": 20,
    "text": ")",
    "type": "punctuation",
    "value": ")",
  },
  Object {
    "end": 22,
    "start": 21,
    "text": "]",
    "type": "punctuation",
    "value": "]",
  },
  Object {
    "end": 23,
    "start": 22,
    "text": ".",
    "type": "punctuation",
    "value": ".",
  },
  Object {
    "end": 24,
    "start": 23,
    "text": "x",
    "type": "name",
    "value": "x",
  },
  Object {
    "end": 25,
    "start": 24,
    "text": "[",
    "type": "punctuation",
    "value": "[",
  },
  Object {
    "end": 26,
    "start": 25,
    "text": "1",
    "type": "num",
    "value": "1",
  },
  Object {
    "end": 27,
    "start": 26,
    "text": "]",
    "type": "punctuation",
    "value": "]",
  },
  Object {
    "end": 28,
    "start": 27,
    "text": ";",
    "type": "punctuation",
    "value": ";",
  },
]
`;
